proposals/NNNN-Continuous-Integration-And-Testing.md

```diff
@@ -0,0 +1,190 @@
+# Continuous integration and testing
+
+* Proposal: [NNNN](NNNN-Continuous-Integration-And-Testing.md)
+* Author: [Pavel Zhdanov](https://github.com/ZhdanovP), [Mykyta Nikora](https://github.com/MNikora)
+* Status: **Review**
+* Impacted Platforms: [Core]
+
+## Introduction
+
+This proposal is about implementation of the Continuous Integration (CI) process for keeping SDL in a trouble-free state. 
+
+## Motivation
+
+Continuous Integration provides a lot of advantages for all actors involved in software product creation such as software developers, product managers, and enterprises.
+
+CI simplifies new features adding, helps to produce high-quality code and makes the development process more predictable by early detection of bugs, regression, and other potential risks.
+
+## Proposed solution
+
+After this proposal is implemented all necessary scripts and environment configuration files, running checks and test results will be ready to use. The main part of the scripts will be used by developers to unify the workstations and work processes. The new repository will be created for storing all configuration needed files. The scripts can be used to deploy the CI on any server, and server resources (cores, memory, disk space, etc.) can be expanded through the cluster tools usage. So, this is what we will get as a result of the implementation, now let us take a look at implementation details.
+
+Currently, the branches that constantly need code health monitoring can be classified into two types:
+
+1. **develop\master branch** - the branch that has to be under permanent testing of the main functionality. There is a potential risk that committed changes can break something. 
+2. **feature branch** - this branch type is used when new feature should be developed and added.
+
+Let's consider a few goals that have to be achieved using CI.
+
+### 1. Continuous testing on a specific branch to maintain the current working condition
+
+The following entities are specific types of checks that should be considered on the develop and feature branches:
+
+- Smoke Tests
+- Regression Tests
+- Feature Tests (applicable only for the feature branches)
+- Unit Tests + Coverage Report
+- Coding Style Check
+- Static Code Analysis
+- Dynamic Analysis on Smoke Tests
+
+**! Note that each of the above checks (except Coding Style Check and Static Code Analysis) should be run for all supported Policies flows (External proprietary, Proprietary, HTTP).**
+
+Also, it would be nice to perform project compilation with the different sets of flags: with and without ENABLE_LOG, ENABLE_SECURITY, DEBUG/RELEASE mode etc - since this allows to find hidden problems.
```

Comment:
```suggestion
Also, it would be nice to perform project compilation with the different sets of flags: with and without ENABLE_LOG, ENABLE_SECURITY, DEBUG/RELEASE mode etc - since this allows us to find hidden problems.
```